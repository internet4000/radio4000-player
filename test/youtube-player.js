import test from 'ava'
import Vue from 'vue'
import Component from '../src/YoutubePlayer.vue'

test.cb('renders', t => {
	const vm = new Vue(Component).$mount()
	const tree = {
		$el: vm.$el.outerHTML
	}
	const $ = selector => vm.$el.querySelectorAll(selector)

	t.snapshot(tree)

	t.is(vm.$el.className, 'ytplayer', 'has an element for the youtube iframe')
	t.is(vm.$props.autoplay, undefined, 'autoplay is not enabled')

	t.is(vm.$data.didPlay, false)
	vm.$props.videoId = '-Op4D4bkK6Y'
	vm.$nextTick(() => {
		t.is(vm.$data.didPlay, true, 'after playing, didPlay changes to true')
		t.end()
	})
})
