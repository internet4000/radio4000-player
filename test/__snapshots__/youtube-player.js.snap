// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "$el": "<div class=\\"ytplayer\\"></div>",
}
`;
